packages/subgraph/schema.graphql

```diff
@@ -1,13 +1,256 @@
+  rewards: BigDecimal!
+  "Total rewards collected from protocol in USD"
+  rewardsUSD: BigDecimal!
+  "Total deposits + total rewards - slashes - total withdrawals"
+  currentPrincipal: BigDecimal!
+  "Current Principle in USD"
+  TVL: BigDecimal!
+
+  "Total amount of protocol fees collected from tenderizer"
+  protocolFees: BigDecimal!
+  "Total amount of protocol fees collected from tenderizer in USD"
+  protocolFeesUSD: BigDecimal!
+  "Total amount of liquidity fees collected from tenderizer"
+  liquidityFees: BigDecimal!
+  "Total amount of liquidity fees collected from tenderizer in USD"
+  liquidityFeesUSD: BigDecimal!
+
+  "Day data for tenderizer"
+  dayData: [TenderizerDay!]! @derivedFrom(field: "tenderizer")
+}
+
+type TenderFarm @entity {
+  "Name of the protocol eg. 'livepeer'"
+  id: ID!
+  "Protocol"
+  protocol: Protocol!
+
+  "Total deposits made into farm"
+  deposits: BigDecimal!
+  "Total withrawals made from farm"
+  withdrawals: BigDecimal!
+  "Total rewards collected from farm"
+  harvest: BigDecimal!
+  "Total rewards collected from farm in USD"
+  harvestUSD: BigDecimal!
+  "Total depsotis - total withdrawals"
+  currentPrincipal: BigDecimal!
+  "Total rewards added"
+  rewards: BigDecimal!
+  "CurrentPrincipal in USD"
+  TVL: BigDecimal!
+
+  "Day data for tenderFarm"
+  dayData: [TenderFarmDay!]! @derivedFrom(field: "tenderFarm")
+}
+
+type TenderizerDay @entity {
+  "<Timestamp rounded to current day by dividing by 86400>_<portocol>"
+  id: ID!
+  "Tenderizer"
+  tenderizer: Tenderizer!
+  "The date beginning at 12:00am UTC"
+  date: Int!
+
+  "deposits volume in tenderizer for the day"
+  deposits: BigDecimal!
+  "withdraw volume in tenderizer for the day"
+  withdrawals: BigDecimal!
+  "staking rewards collected from protocol for the day"
+  rewards: BigDecimal!
+
+  "Principle on the first event of the day"
+  startPrinciple: BigDecimal!
+  "Day APY - rewards/principle %"
+  APY: BigDecimal!
+}
+
+type TenderFarmDay @entity {
+  "<Timestamp rounded to current day by dividing by 86400>_<portocol>"
+  id: ID!
+  "TenderFarm"
+  tenderFarm: TenderFarm!
+  "The date beginning at 12:00am UTC"
+  date: Int!
+
+  "farm volume for the day"
+  deposits: BigDecimal!
+  "unfarm volume for the day"
+  withdrawals: BigDecimal!
+  "rewards added for the day"
+  rewards: BigDecimal!
+  "harvest volume for the day"
+  harvest: BigDecimal!
+
+  "principle in Token (not BPT) value at the start of the day"
+  startPrinciple: BigDecimal!
+  "Day APY - total rewards / total volume %"
+  APY: BigDecimal!
+}
+
+type User @entity {
+  "Wallet address of user"
+  id: ID!
+  "Data for each integration per user"
+  protocols: [UserProtocol!]! @derivedFrom(field: "user")
+}
+
+type UserProtocol @entity { 
+  "<wallet-address>_<portocol>"
+  id: ID!
+  "Protocol"
+  protocol: Protocol!
+  "User"
+  user: User!
+
+  "Users current stake: total deposits - withdrawals"
+  tenderizerStake: BigDecimal! # TODO: Better name?
```

Comment:
Based on this we can actually calculate rewards for a user as well

current tender token balance - tenderizerStake

Reply:
Hmm, thing is the user need not have all the tenderTokens in his wallet and it might be transferred elsewhere, so might not be the best idea to rely on current tender token balance

Reply:
It's not the current , it's the balance at the time the event gets emitted. So even if the tokens are transferred elsewhere (which we might need to account for regardless because this would be the same as an unstake really) this would still show the correct total rewards up until the point of the transfer.